src/inference/Inference.jl

```diff
@@ -356,13 +357,15 @@ function AbstractMCMC.bundle_samples(
     parray = MCMCChains.concretize(parray)
 
     # Chain construction.
-    return MCMCChains.Chains(
+    chain = MCMCChains.Chains(
         parray,
         nms,
         (internals = extra_params,);
         evidence=le,
         info=info,
-    ) |> sort
+    )
+
+    return sort_chain ? sort(chain) : chain
```

Comment:
Ah I think I remember why it was removed - doesn't this create a type instability? IIRC `sample` can't be inferred due to limitations of Julia's logging system anyway, so I guess it doesn't matter :shrug:

Reply:
Ah, but do we care? :confused:

Reply:
As I said, I am pretty sure it's unstable either way, so one should always use a function barrier if it's important. Hence I'm fine with the change :slightly_smiling_face: